### ISSUE TITLE
key_size for hmac-sha256@ssh.com correct?

### ISSUE THREAD

**Issue by norrisjeremy** [ronf/asyncssh#399]
Is the key_size for `hmac-sha256@ssh.com` deliberately defined as 16 instead of 32 in mac.py?
```
    (b'hmac-sha256@ssh.com',     16, 32, False, _HMAC,    (sha256,),  True),
```

**Comment by ronf** [comment]
Yes - this is correct, based on my interoperability testing against SSH.com's Tectia SSH client and server. From what I was able to tell, they first defined "hmac-sha256@ssh.com" with a 128 bit key and 256 bit hash and then later added "hmac-sha256-2@ssh.com" with a 256 bit key and hash.

Note that both of these MAC algorithms are independent of the more widely supported "hmac-sha2-256", which also has a 256 bit key and hash. That's probably the one you'll want to use in almost all cases. I just added support for some of the ssh.com algorithms to maximize AsyncSSH's interoperability.

**Comment by norrisjeremy** [comment]
Ok, that makes sense. I've made several contributions to a fork of the [JSch library](https://github.com/mwiede/jsch) and was evaluating the possibility of adding support for the various `...@ssh.com` algorithms into it to increase interoperability (similar to what you have done with AsyncSSH).

**Comment by norrisjeremy** [comment]
A bit unrelated but, I noticed that the following Kex algorithms aren't mentioned in the docs: is that intentional?
- diffie-hellman-group-exchange-sha224@ssh.com
- diffie-hellman-group-exchange-sha384@ssh.com
- diffie-hellman-group-exchange-sha512@ssh.com

**Comment by ronf** [comment]
Good catch! This is in fact a mistake. All of these algorithms are disabled by default, but they should have been added to the section of the documentation listing key exchange algorithms which are supported but initially disabled.

**Comment by ronf** [comment]
This is fixed in commit 6ecd296 in the "develop" branch and will be included in the next release.

**Comment by ronf** [comment]
This change is now available in AsyncSSH version 2.7.1.
